salt/metalk8s/kubeadm/init/kubelet-start/configured.sls

```diff
@@ -0,0 +1,51 @@
+{%- from "metalk8s/map.jinja" import kubelet with context %}
+
+Create kubelet service environment file:
+  file.managed:
+    - name: {{ kubelet.service.environment_file }}
+    - source: salt://metalk8s/kubeadm/init/kubelet-start/files/kubeadm.env
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - context:
+        options: {{ kubelet.service.options }}
+
+Create kubelet config file:
+  file.serialize:
+    - name: {{ kubelet.config_file }}
+    - user: root
+    - group: root
+    - mode: 644
+    - formatter: yaml
+    - dataset: {{ kubelet.config }}       
+
+{%- set kubelet_service_config = salt['file.join'](
+          kubelet.service.config_path,
+          kubelet.service.config_name) %}
+Configure kubelet service:
+  file.managed:
+    - name: {{ kubelet_service_config }}
+    - source: salt://metalk8s/kubeadm/init/kubelet-start/files/service-kubelet-{{ grains['init'] }}.conf
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+    - context:
+        env_file: {{ kubelet.service.environment_file }}
+        conf_file: {{ kubelet.config_file }}
+    - require:
+      - file: Create kubelet service environment file
+      - file: Create kubelet config file
+
+Start and enable kubelet:
+  service.running:
+    - name: kubelet
+    - enable: True
+    - watch:
```

Comment:
I think "onchanges" is better here.
"watch" would trigger the restart whatever happens.

Reply:
Only if these file are modified so btw it's what we want I think no ?

Don't know if it's also do like a `systemctl daemon-reload`

Reply:
After rereading the doc, you're right

Reply:
You need `daemon-reload` when the unit and/or `EnvironmentFile` gets updated, not when the `kubelet` configuration file gets updated. So yes, you need it.

Reply:
Indeed I will add it

Reply:
Ok no, it's not needed already done by salt since 29 Sep 2016 https://github.com/saltstack/salt/pull/36538